src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -1837,23 +1837,15 @@ private IEnumerable<PSModuleInfo> CreateFakeModuleObject(IEnumerable<ModuleSpeci
             else if ((requiredProcessorArchitecture != ProcessorArchitecture.None) &&
```

Comment:
@iSazonov, your last commit had 1 failures in `PowerShell-CI-linux`
New-ModuleManifest tests.New-ModuleManifest works with assembly architecture: Amd64
```powershell
Cannot add type. Compilation errors occurred.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Module/NewModuleManifest.Tests.ps1: line 173
173:         Add-Type -TypeDefinition $a -CompilerOptions "/platform:$roslynArch" -OutputAssembly $assemblyPath

```